mlserver/server.py

```diff
@@ -54,7 +56,18 @@ def __init__(self, settings: Settings):
             on_model_reload=on_model_reload,  # type: ignore
             on_model_unload=on_model_unload,  # type: ignore
         )
-        self._model_repository = ModelRepository(self._settings.model_repository_root)
+
+        if model_repository:
+            self._model_repository = model_repository
+        elif self._settings.model_repository_implementation:
+            self._model_repository = self._settings.model_repository_implementation(
+                self._settings.model_repository_implementation_args
```

Comment:
Given that `model_repository_implementation_args` is a dict, should we just unpack it and pass it as kwargs (i.e. with the `**` thingy)?

Reply:
I like it!, thanks.

Reply:
Done